### PR TITLE
String and Cow on `no_std`

### PR DESCRIPTION
**Pull request by bkchr**
@gavofyork could you also please release the new version to crates.io?

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -32,6 +32,8 @@ extern crate arrayvec;
 pub mod alloc {
 	pub use std::boxed;
 	pub use std::vec;
+    pub use std::string;
```
Comment (sorpaas) [inline src/lib.rs]: Should use `<tab>` for indentation here.

File: src/codec.rs [hunk 2]
```diff
@@ -258,36 +260,32 @@ impl<'a> Encode for &'a str {
...
+impl<'a, T: ToOwned + ?Sized> Decode for Cow<'a, T> where
 	<T as ToOwned>::Owned: Decode
 {
 	fn decode<I: Input>(input: &mut I) -> Option<Self> {
-		Some(::std::borrow::Cow::Owned(Decode::decode(input)?))
+		Some(Cow::Owned(Decode::decode(input)?))
 	}
 }
 
-#[cfg(feature = "std")]
```
Comment (gavofyork) [inline src/codec.rs]: There's a reason for this: we don't want `String` in wasm environments.

**Comment by gavofyork** [comment]
There's a reason these things were disabled in `no_std` - we don't want to expose them in Wasm.

File: src/codec.rs (hunk 2 above)
Comment (pepyakin) [reply src/codec.rs]: But what the reason exactly?

**Pull request closed without merging**